File: codes/monorepo-microservice-example/user-service/src/main/java/vduczz/userservice/application/service/AuthService.java
package vduczz.userservice.application.service;

import vduczz.userservice.application.port.in.dto.request.AuthRequestDto;
import vduczz.userservice.application.port.in.dto.response.AuthResponseDto;

public interface AuthService {
    AuthResponseDto register(AuthRequestDto.RegisterRequest request, boolean mq, boolean kafka);
}

File: codes/mini-project/src/main/kotlin/com/vduczz/mini_project/service/AuthService.kt
package com.vduczz.mini_project.service

import com.vduczz.mini_project.core.command.CreateUserCommand
import com.vduczz.mini_project.core.command.LoginCommand
import com.vduczz.mini_project.dto.response.AuthResponse

interface AuthService {

    fun register(command: CreateUserCommand): AuthResponse

    fun login(credential: LoginCommand): AuthResponse
}